packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -137,7 +137,7 @@ async function instantiateModule(
     if (dep[0] !== '.' && dep[0] !== '/') {
       return nodeImport(dep, mod.file!, resolveOptions)
     }
-    dep = unwrapId(dep)
+    dep = unwrapId(dep).replace(NULL_BYTE_PLACEHOLDER, '\0')
```

Comment:
IIUC, this may be the only line needed to do the fix. And it isn't needed because we call `ssrLoadModule`, but because we check `pendingImports` and `moduleGraph.urlToModuleMap.get(dep)`, and both should be in the rollup URL space.